ipam.go

```diff
@@ -5,39 +5,69 @@ import (
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	NewPrefix(ctx context.Context, cidr string) (*Prefix, error)
 	// DeletePrefix delete a Prefix from a string notation.
 	// If the Prefix is not found an NotFoundError is returned.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	DeletePrefix(ctx context.Context, cidr string) (*Prefix, error)
 	// AcquireChildPrefix will return a Prefix with a smaller length from the given Prefix.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	AcquireChildPrefix(ctx context.Context, parentCidr string, length uint8) (*Prefix, error)
 	// AcquireSpecificChildPrefix will return a Prefix with a smaller length from the given Prefix.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	AcquireSpecificChildPrefix(ctx context.Context, parentCidr, childCidr string) (*Prefix, error)
 	// ReleaseChildPrefix will mark this child Prefix as available again.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	ReleaseChildPrefix(ctx context.Context, child *Prefix) error
 	// PrefixFrom will return a known Prefix.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	PrefixFrom(ctx context.Context, cidr string) *Prefix
 	// AcquireSpecificIP will acquire given IP and mark this IP as used, if already in use, return nil.
 	// If specificIP is empty, the next free IP is returned.
 	// If there is no free IP an NoIPAvailableError is returned.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	AcquireSpecificIP(ctx context.Context, prefixCidr, specificIP string) (*IP, error)
 	// AcquireIP will return the next unused IP from this Prefix.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	AcquireIP(ctx context.Context, prefixCidr string) (*IP, error)
 	// ReleaseIP will release the given IP for later usage and returns the updated Prefix.
 	// If the IP is not found an NotFoundError is returned.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	ReleaseIP(ctx context.Context, ip *IP) (*Prefix, error)
 	// ReleaseIPFromPrefix will release the given IP for later usage.
 	// If the Prefix or the IP is not found an NotFoundError is returned.
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
 	ReleaseIPFromPrefix(ctx context.Context, prefixCidr, ip string) error
 	// Dump all stored prefixes as json formatted string
+	// This operation is scoped to the root namespace unless a different namespace is provided in the context.
```

Comment:
So with this we loose the ability to dump the whole ipam db in a database agnostic format, we must call that with all namespaces now. I think we should consider either have a DumpAll or a DumpNamespace, WDYT ?